### PR TITLE
Remove custom arguments in object.__new__ of BaseInstrumentor

### LINKED ISSUE #1438: System metrics instrumentation not working with custom defined configuration
System metric instrumentation is not functional if configuration on which metrics to be exported is explicitly provided. As a minimal example, this code

```python
from opentelemetry.metrics import set_meter_provider
from opentelemetry.instrumentation.system_metrics import SystemMetricsInstrumentor
from opentelemetry.sdk.metrics import MeterProvider
from opentelemetry.sdk.metrics.export import (
    ConsoleMetricExporter,
    PeriodicExportingMetricReader,
)

exporter = ConsoleMetricExporter()
set_meter_provider(MeterProvider([PeriodicExportingMetricReader(exporter)]))

configuration = {
    "runtime.memory": ["rss", "vms"],
    "runtime.cpu.time": ["user", "system"],
}

SystemMetricsInstrumentor(config=configuration).instrument()
```

results in

```
Traceback (most recent call last):
  File ".../test.py", line 15, in <module>
    SystemMetricsInstrumentor(config=configuration).instrument()
  File ".../lib/python3.10/site-packages/opentelemetry/instrumentation/instrumentor.py", line 51, in __new__
    cls._instance = object.__new__(cls, *args, **kwargs)
TypeError: object.__new__() takes exactly one argument (the type to instantiate)
```

I am happy to look into fixing this. Removing `*args` and `**kwargs` in `opentelemetry/instrumentation/instrumentor.py:51` actually solves the issue here but I'd like to understand the implications as this implies changing the interface class.

### PR DESCRIPTION
**Pull request by rbagd**
# Description

Classes inheriting from `BaseInstrumentor` class are singletons tracked by `_instance` attribute and this piece of code

```
    def __new__(cls, *args, **kwargs):
        if cls._instance is None:
            cls._instance = object.__new__(cls, *args, **kwargs)
```

If an instrumentor child class with parameters, e.g. `SystemMetricsInstrumentor` is initialized first, then these parameters are also passed to `object.__new__` which results in an error as in #1438.

```
# This works
SystemMetricsInstrumentor()
SystemMetricsInstrumentor(config={})
```

```
# This does not
SystemMetricsInstrumentor(config={})
SystemMetricsInstrumentor()
```

Given [the answers in this SO question](
https://stackoverflow.com/questions/59217884/new-method-giving-error-object-new-takes-exactly-one-argument-the-typ), I am inclined to think that it's not needed to pass arguments to `object.__new__` and they can be safely removed. Those arguments will still end up in the `__init__` call.

Fixes #1438

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Added a unit test in system metrics to reproduce the issue (this requires singleton class to be reset during test run)
- [X] Additional tests were ran with `tox`for multiple other instrumented packages

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by ocelotl** [comment]
Nice, please add another test case, one that pretty much does this and checks no exception is raised:

``` python
def test_some_test_case_name(self):
    try:
        SystemMetricsInstrumentor()
        SystemMetricsInstrumentor(config={})
    except Exception as error:
        self.fail(f"Unexpected exception {error} raised")
    # do any necessary resetting
        try:
        SystemMetricsInstrumentor()
        SystemMetricsInstrumentor(config={})
    except Exception as error:
        self.fail(f"Unexpected exception {error} raised")
```

**Comment by rbagd** [comment]
I added an extra case as per @ocelotl request. I assumed he meant the init order should be inverted in the second case. :wink:

**Comment by ocelotl** [comment]
> I added an extra case as per @ocelotl request. I assumed he meant the init order should be inverted in the second case. wink

right! :sweat_smile: some tests are failing, please take a look

**Comment by rbagd** [comment]
All failing tests I've checked were related to #1460. Now that it got merged, I think it should be alright.

**Comment by rbagd** [comment]
Both failing jobs failed due to some issue with `git` itself. Is it possible to re-run just those two?

```
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +95327fbe060f4136c197c59869fb13484d3b77c8:refs/remotes/pull/1439/merge
Error: fatal: protocol error: bad pack header
Error: The process '/usr/bin/git' failed with exit code 128
```

**Comment by srikanthccv** [comment]
>Both failing jobs failed due to some issue with git itself. Is it possible to re-run just those two?

This should be fine now.

**Comment by srikanthccv** [comment]
@ocelotl PTAL

**Comment by rbagd** [comment]
@ocelotl @srikanthccv Linting failed due to a style change in `black`. I think I got it fixed so if possible to get this merged before they update it again, it'd be great. :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -102,6 +102,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1424](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1424))
 - Remove db.name attribute from Redis instrumentation
   ([#1427](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1427))
+- `opentelemetry-instrumentation-system-metrics` Fix initialization of the instrumentation class when configuration is provided
```
Comment (srikanthccv) [inline CHANGELOG.md:105]: This should be in ## Unreleased >> ## Fixed section

Comment (rbagd) [reply CHANGELOG.md]: Good point! Updated.

**Comment by srikanthccv** [comment]
@ocelotl PTAL

**Pull request merged**